### PR TITLE
feat: Add golangci-lint for tests

### PR DESCRIPTION
**Pull request by 0xFelix**
**What this PR does / why we need it**:

- Add golangci-lint for tests and borrow .golangci.yaml from kubevirt/kubevirt.
- Apply golangci-lint suggestions to tests
- Prevents errors like resourceVersion being already set, when trying
to create a new VM by not reusing VMs in functional tests
- Rename check.sh to check-tree-clean.sh, remove instancetype version check from it and add new target to Makefile.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

File: tests/functests/instancetype_test.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ import (
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/rand"
-	"k8s.io/utils/pointer"
+	"k8s.io/utils/ptr"
```
Comment (ksimon1) [inline tests/functests/instancetype_test.go:14]: why is this change necessary?

Comment (0xFelix) [reply tests/functests/instancetype_test.go:14]: Because `k8s.io/utils/pointer` is deprecated and `golangci-lint` complains about it.

**Comment by 0xFelix** [comment]
Blocked by https://github.com/kubevirt/project-infra/pull/3109
